src/test/java/io/kestra/plugin/aws/kinesis/PutRecordsTest.java

```diff
@@ -0,0 +1,269 @@
+package io.kestra.plugin.aws.kinesis;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.plugin.aws.kinesis.model.Record;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import jakarta.inject.Inject;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.jackson.Jacksonized;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.KinesisClientBuilder;
+import software.amazon.awssdk.services.kinesis.model.*;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.List;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@MicronautTest
+@Testcontainers
+class PutRecordsTest {
+    private static final ObjectMapper MAPPER = JacksonMapper.ofIon()
+        .setSerializationInclusion(JsonInclude.Include.ALWAYS);
+
+    protected static LocalStackContainer localstack;
+    @Inject
+    protected RunContextFactory runContextFactory;
+    @Inject
+    protected StorageInterface storageInterface;
+
+    @BeforeAll
+    static void startLocalstack() throws IllegalVariableEvaluationException, InterruptedException {
+        localstack = new LocalStackContainer(DockerImageName.parse("localstack/localstack:1.3.1"));
```

Comment:
I think you used 1.4.0 on the secret manager lib?
Maybe you can try with this version and refactor all tests to share a constant somewhere to use the same localstack version (not mandatory).

Reply:
I'll do that in a separate pr to use the same version everywhere as we disccused on slack :+1: